enhancements/monitoring/monitoring-config.md

```diff
@@ -0,0 +1,305 @@
+* Custom resources play better with GitOps workflows. 
+* CRDs supports multiple actors managing the same resource which is a key property for the Observability service of Advanced Cluster Management.
+
+### User Stories
+
+As a cluster administrator, I want to be able to add a CRD so that I can configure monitoring stack.
+
+
+### Goals
+
+- Replace the existing ConfigMaps with CRDs.
+- Automated and friction-less upgrade for users.
+
+### Non-Goals
+
+- Make things more complicated and easy to break.
+- Change any behavior of monitoring stack
+
+## Proposal
+
+Currently in CMO, a config map provides a way to inject configuration data into pods. There are two configmaps for the different stacks:
+
+    cluster-monitoring-config: Default platform monitoring components. A set of platform monitoring components are installed in the openshift-monitoring project by default during an OpenShift Container Platform installation. This provides monitoring for core cluster components including Kubernetes services. The default monitoring stack also enables remote health monitoring for clusters.
+
+    user-workload-monitoring-config: Components for monitoring user-defined projects. After optionally enabling monitoring for user-defined projects, additional monitoring components are installed in the openshift-user-workload-monitoring project. This provides monitoring for user-defined projects. These components are illustrated in the User section in the following diagram.
+
+
+Two distinct CRDs are necessary because they are managed by different personas with specific roles and responsibilities:
+
+    - UWM admins: manage the configuration of the UWM components (edit permissions on the openshift-user-workload-monitoring/user-workload-monitoring-config ConfigMap).
+    - Cluster admins: manage the configuration of the Platform monitoring components.
+
+In managed OpenShift clusters like OSD/ROSA, two separate CRDs are necessary because platform SREs manage the cluster's platform monitoring stack, while customers manage the user-defined monitoring stack. This separation ensures that each group maintains control over their specific monitoring configurations, reducing conflicts and enhancing system management.
+
+[More info](https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/multi-tenant-alerting.md)
+
+
+- Replace configmaps with CRD:
+
+Current Config struct in CMO
+  
+```
+  type Config struct {
+	Images                               *Images `json:"-"`
+	RemoteWrite                          bool    `json:"-"`
+	CollectionProfilesFeatureGateEnabled bool    `json:"-"`
+
+	ClusterMonitoringConfiguration *ClusterMonitoringConfiguration `json:"-"`
+	UserWorkloadConfiguration      *UserWorkloadConfiguration      `json:"-"`
+}
+```
+
+We will strive to maintain the previous structure as much as possible while adapting it to OpenShift API standards. 
+
+Current ClusterMonitoringConfiguration struct:
```

Comment:
can we include a perma link here?

Reply:
https://github.com/openshift/cluster-monitoring-operator/blob/eca25a9571d721a90f1b7f8d26931c679ae93aec/pkg/manifests/types.go#L36